frontend/public/components/overview.jsx

```diff
@@ -0,0 +1,372 @@
+    const {filteredItems, filterValue, groupedItems, groupOptions, selectedGroupLabel} = this.state;
+
+    return <div className="co-m-pane">
+      <OverviewToolbar
+        filterValue={filterValue}
+        groupOptions={groupOptions}
+        handleFilterChange={this.handleFilterChange}
+        handleGroupChange={this.handleGroupChange}
+        selectedGroup={selectedGroupLabel}
+      />
+      <div className="co-m-pane__body">
+        <StatusBox data={filteredItems} loaded={loaded} loadError={loadError} label="Resources">
+          <ProjectOverview groups={groupedItems} />
+        </StatusBox>
+      </div>
+    </div>;
+  }
+}
+
+OverviewDetails.displayName = 'OverviewDetails';
+
+OverviewDetails.propTypes = {
+  deploymentConfigs: PropTypes.object,
+  deployments: PropTypes.object,
+  loaded: PropTypes.bool,
+  loadError: PropTypes.object,
+  pods: PropTypes.object,
+  replicationControllers: PropTypes.object,
+  replicaSets: PropTypes.object,
+  statefulSets: PropTypes.object,
+};
+
+export const Overview = ({namespace}) => {
+  const resources = [
+    {
+      isList: true,
+      kind: 'Pod',
+      namespace,
+      prop: 'pods'
+    },
+    {
+      isList: true,
+      kind: 'ReplicationController',
+      namespace,
+      prop: 'replicationControllers'
+    },
+    {
+      isList: true,
+      kind: 'DeploymentConfig',
+      namespace,
+      prop: 'deploymentConfigs'
+    },
+    {
+      isList: true,
+      kind: 'Deployment',
+      namespace,
+      prop: 'deployments'
+    },
+    {
+      isList: true,
+      kind: 'ReplicaSet',
+      namespace,
+      prop: 'replicaSets'
+    },
+    {
+      isList: true,
+      kind: 'StatefulSet',
+      namespace,
+      prop: 'statefulSets'
+    }
+  ];
+  return <div className="overview">
+    <Firehose resources={resources}>
+      <OverviewDetails />
+    </Firehose>
+  </div>;
+};
+
+Overview.displayName = 'Overview';
+
+Overview.propTypes = {
+  namespace: PropTypes.string.isRequired,
+};
+
+
+class OverviewPage_ extends React.PureComponent {
+  render() {
+    const {match, flags} = this.props;
+    const namespace = _.get(match, 'params.ns');
+    const fake = flags.OPENSHIFT && !flags.PROJECTS_AVAILABLE;
+
+    if (flagPending(flags.OPENSHIFT) || flagPending(flags.PROJECTS_AVAILABLE)) {
+      return null;
+    }
+
+    return <React.Fragment>
+      <StartGuide dismissible={true} style={{margin: 15}} />
+      <Helmet>
+        <title>Project Overview</title>
+      </Helmet>
+      <NavTitle title="Project Overview" />
+      <Overview
+        fake={fake} //TODO implement first time user experience using this prop as a flag
+        namespace={namespace}
+      />
+    </React.Fragment>;
+  }
+}
+
+OverviewPage_.displayName = 'OverviewPage';
+
+OverviewPage_.propTypes = {
+  match: PropTypes.object.isRequired,
+  flags: PropTypes.object.isRequired
+};
+
+export const OverviewPage = connectToFlags(FLAGS.OPENSHIFT, FLAGS.PROJECTS_AVAILABLE)(OverviewPage_);
```

Comment:
One drawback to this is you can't have private components. You have to export them all. It becomes more difficult to tell what's used internally and what's meant to be reused. It also discourages breaking things into small, simple components because it's a lot more work.

This would be a large refactoring effort and make it harder to backport fixes to previous releases as well because everything has moved. If you navigate to symbols instead of files in your editor, I feel like it shouldn't matter much what file the component is in.